src/adapters/scene_direction.js

```diff
@@ -73,8 +74,11 @@ export default class SceneDirection {
       if (isActive) {
         mainRoute = route;
       }
-      this.map.setFeatureState({ source: `source_${route.id}`, id: 1 }, { isActive });
+      const geoJson = this.getRouteGeoJson(route.geometry, isActive);
+      this.map.getSource(`source_${route.id}`).setData(geoJson);
```

Comment:
I understand that using `setFeatureState` is not practical to update multiple features. 
But reloading the whole geoJson data seems to be significantly slower. 

Maybe another solution could be to define a routeId property in the features, and update the layer style (to set the `line-color` rule dynamically) ?

Reply:
Thanks for the suggestion! :ok_hand: 
I did it with the second alternative, as it didn't require manipulating each feature independently, requires a lot less code, and we already did something similar for the pedestrian case.
Now it's generic and snappy :)